### PR TITLE
Fix bugs with OIDC and SAML forms

### LINKED ISSUE #1382: Unable to delete OpenID and SAML providers configuration
Version: 0.7.6

I have three SAML configurations that I added to test out SAML integration with JumpCloud which is failing for some reason which I am still trying to investigate. However, I am unable to delete the configurations and noticed that the error in the logs points to another SAML configuration which I am not attempting to delete.

```
17:32:32.729 [error] GenServer #PID<0.2328.0> terminating
** (Ecto.CastError) expected params to be a :map, got: `%FzHttp.Configurations.Configuration.SAMLIdentityProvider{id: "XXXX-azuread", label: "Azure-AD", base_url: "https://XXXX.XXXX.XXXX/auth/saml", metadata: "<?xml version=\"1.0\" encoding=\"utf-8\"?><EntityDescriptor ID=\"1234\" entityID=\"1234\" xmlns=\"urn:oasis:names:tc:SAML:2.0:metadata\"><Signature xmlns=\"1234\"></Signature ></EntityDescriptor>", sign_requests: false, sign_metadata: false, signed_assertion_in_resp: false, signed_envelopes_in_resp: false, auto_create_users: false}`
    (ecto 3.9.4) lib/ecto/changeset.ex:498: Ecto.Changeset.cast/4
    (fz_http 0.7.6) lib/fz_http/configurations/configuration/saml_identity_provider.ex:29: FzHttp.Configurations.Configuration.SAMLIdentityProvider.changeset/2
    (ecto 3.9.4) lib/ecto/changeset/relation.ex:137: Ecto.Changeset.Relation.do_cast/6
    (ecto 3.9.4) lib/ecto/changeset/relation.ex:356: Ecto.Changeset.Relation.map_changes/9
    (ecto 3.9.4) lib/ecto/changeset/relation.ex:121: Ecto.Changeset.Relation.cast/5
    (ecto 3.9.4) lib/ecto/changeset.ex:832: Ecto.Changeset.cast_relation/4
    (fz_http 0.7.6) lib/fz_http/configurations/configuration.ex:83: FzHttp.Configurations.Configuration.changeset/2
    (fz_http 0.7.6) lib/fz_http/configurations.ex:70: FzHttp.Configurations.update_configuration/2
Last message: %Phoenix.Socket.Message{topic: "lv:phx-Fz3sIxyQZ0BAOQGk", event: "event", payload: %{"event" => "delete", "type" => "click", "value" => %{"key" => "XXXX-jumpcloud1", "type" => "saml", "value" => ""}}, ref: "23", join_ref: "21"}
```

I tried clean up the configuration for the configuration that the above log is erroring on and have left the minimum required data. However, unable to delete still.

### PR DESCRIPTION
**Pull request by AndrewDryga**
1. When OIDC/SAML is deleted the state of the LV socket becomes invalid and basically makes it looks like everything was deleted
2. When there is more than one OIDC/SAML config trying to delete it leads to a crash
3. Updates of OIDC/SAML were messy and hard to follow, they are reworked and more tests are added around them

Closes #1382

### CODE REVIEW

File: apps/fz_http/lib/fz_http_web/live/setting_live/oidc_form_component.ex [hunk 1]
```diff
@@ -174,55 +175,50 @@ defmodule FzHttpWeb.SettingLive.OIDCFormComponent do
...
-    case update do
-      {:ok, _config} ->
-        {:noreply,
-         socket
-         |> put_flash(:info, "Updated successfully.")
-         |> redirect(to: socket.assigns.return_to)}
-
-      {:error, changeset} ->
-        {:noreply, assign(socket, :changeset, changeset)}
+    create_changeset = Configurations.Configuration.OpenIDConnectProvider.create_changeset(params)
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http_web/live/setting_live/oidc_form_component.ex:190]: Will this create duplicate config entries?

Comment (AndrewDryga) [reply apps/fz_http/lib/fz_http_web/live/setting_live/oidc_form_component.ex:190]: Nope, its just a hack to validate params which was already there before PR

Comment (jamilbk) [reply apps/fz_http/lib/fz_http_web/live/setting_live/oidc_form_component.ex:190]: KK 👍
